nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp

```diff
@@ -74,16 +84,29 @@ inline BT::NodeStatus GoalUpdater::tick()
     }
   }
 
+  if (!last_goals_received_.empty()) {
+    goals = last_goals_received_;
+  }
+
   setOutput("output_goal", goal);
+  setOutput("output_goals", goals);
   return child_node_->executeTick();
 }
 
 void
 GoalUpdater::callback_updated_goal(const geometry_msgs::msg::PoseStamped::SharedPtr msg)
 {
+  std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
No mutex is needed since we spin in the `tick` function -- we know its not executed in parallel to access